src/main/scala/is/hail/expr/Type.scala

```diff
@@ -22,39 +22,30 @@ import scala.reflect.classTag
 abstract class BaseType
 
 object Type {
-  val genOptionalScalar: Gen[Type] = Gen.oneOf[Type](TBooleanOptional, TInt32Optional, TInt64Optional, TFloat32Optional, TFloat64Optional, TStringOptional,
-    TVariant(GenomeReference.GRCh37, false), TAltAlleleOptional, TGenotypeOptional, TLocus(GenomeReference.GRCh37, false), TInterval(GenomeReference.GRCh37, false), TCallOptional)
+  def genScalar(required: Boolean) =
+    Gen.oneOf(TBoolean(required), TInt32(required), TInt64(required), TFloat32(required),
+      TFloat64(required), TString(required), TAltAllele(required), TGenotype(required), TCall(required))
 
-  val genRequiredScalar: Gen[Type] = Gen.oneOf[Type](TBooleanRequired, TInt32Required, TInt64Required, TFloat32Required, TFloat64Required, TStringRequired,
-    TVariant(GenomeReference.GRCh37, true), TAltAlleleRequired, TGenotypeRequired, TLocus(GenomeReference.GRCh37, true), TInterval(GenomeReference.GRCh37, true), TCallRequired)
+  val genOptionalScalar = genScalar(false)
 
-  val genScalar: Gen[Type] = Gen.coin(0.2).flatMap( r => if (r) genRequiredScalar else genOptionalScalar )
+  val genRequiredScalar = genScalar(true)
 
-  def genSized(size: Int): Gen[Type] = {
-    if (size < 1)
-      Gen.const(TStruct.empty)
-    else if (size < 2)
-      genScalar
-    else {
-      val genMaybeRequired: Gen[Type] = for {
-        t <- genArb.resize(size - 1)
-        req <- Gen.coin(0.2)
-      } yield if (req) !t else t
-      Gen.oneOfGen(genScalar,
-        genScalar,
-        genScalar,
-        genScalar,
-        genMaybeRequired.map { TArray(_) },
-        genMaybeRequired.map { TSet(_) },
-        Gen.zip(genArb.resize(size - 1), genMaybeRequired).map { case (k, v) => TDict(k, v) },
-        genStruct.resize(size))
-    }
+  def genComplexType(required: Boolean) = {
+    val grDependents = GenomeReference.references.values.flatMap(gr =>
+      Array(TVariant(gr, required), TLocus(gr, required), TInterval(gr, required)))
+    val others = Array(
+      TAltAllele(required), TGenotype(required), TCall(required))
+    Gen.oneOfSeq(grDependents ++ others)
```

Comment:
I'm not really sure what's happening here, but it's apparently returning an Iterable when the function expects a Seq.

Reply:
blah